### PR TITLE
fix: no commit in released versions to report

### LINKED ISSUE #1246: [Bug]: Error in __init__ file when importing pypi topostats release
### Checklist

- [x] Find the offending file in the output. If processing halts, re-run analysis with `topostats --core 1 process`.
- [x] Describe the bug.
- [x] Include the configuration file.
- [x] Copy of the log-file from running with `topostats --log-level debug <command>`.
- [x] The exact command that failed. This is what you typed at the command line, including any options.
- [x] TopoStats version, this is reported by `topostats --version`
- [x] Operating System and Python Version

### Describe the bug

Image
I'm getting this error when trying to import the latest pypi release of napari into topostats

### Copy of the log-file from running with `topostats --log-level debug <command>`

N/A

### Include the configuration file

N/A

### To Reproduce

Failed to import topostats into napari
Also failed with same error when running just "topostats"

### TopoStats Version

Git main branch

### Python Version

3.11

### Operating System

Windows

### Python Packages

absl-py==2.3.1
AFMReader @ git+https://github.com/AFM-SPM/AFMReader@29e486461a0a2b1c860faaf1b4b6df3b5a484432
aiohappyeyeballs==2.6.1
aiohttp==3.13.2
aiosignal==1.4.0
alabaster==1.0.0
annotated-types==0.7.0
app-model==0.4.0
appdirs==1.4.4
art==6.5
asttokens==3.0.0
astunparse==1.6.3
attrs==25.4.0
babel==2.17.0
bermuda==0.1.6
biopython==1.86
build==1.3.0
cachetools==6.2.1
cachey==0.2.1
certifi==2025.10.5
chardet==5.2.0
charset-normalizer==3.4.4
cheap_repr==0.5.2
click==8.3.0
cloudpickle==3.1.1
colorama @ file:///C:/ci_311/colorama_1676422310965/work
comm==0.2.3
contourpy==1.3.3
crc32c==2.8
cycler==0.12.1
dask==2025.10.0
debugpy==1.8.17
decorator==5.2.1
distlib==0.4.0
docstring_parser==0.17.0
docutils==0.21.2
donfig==0.8.1.post1
et_xmlfile==2.0.0
executing==2.2.1
filelock==3.20.0
flatbuffers==25.9.23
flexcache==0.3
flexparser==0.4
fonttools==4.60.1
freetype-py==2.5.1
frozenlist==1.8.0
fsspec==2025.10.0
gast==0.6.0
google-pasta==0.2.0
grpcio==1.76.0
h5py==3.15.1
HeapDict==1.0.1
hsluv==5.0.4
idna==3.11
igor2==0.5.12
imageio==2.37.0
imagesize==1.4.1
importlib_metadata==8.7.0
in-n-out==0.2.1
iniconfig @ file:///C:/b/abs_70kaor_ccd/croot/iniconfig_1752653749904/work
ipykernel==6.31.0
ipython==8.37.0
ipython_pygments_lexers==1.1.1
jedi==0.19.2
Jinja2==3.1.6
joblib==1.5.2
jsonschema==4.25.1
jsonschema-specifications==2025.9.1
jupyter_client==8.6.3
jupyter_core==5.9.1
keras==3.12.0
kiwisolver==1.4.9
lazy_loader==0.4
libclang==18.1.1
llvmlite==0.45.1
locket==1.0.0
loguru==0.7.3
magicgui==0.10.1
Markdown==3.9
markdown-it-py==4.0.0
MarkupSafe==3.0.3
matplotlib==3.10.7
matplotlib-inline==0.2.1
mdurl==0.1.2
ml_dtypes==0.5.3
multidict==6.7.0
namex==0.1.0
napari==0.6.6
napari-afmreader @ git+https://github.com/AFM-SPM/napari-AFMReader.git@58c77c89d732487fd78bffd79fc1444271b27c45
napari-console==0.1.4
napari-plugin-engine==0.2.0
napari-plugin-manager==0.1.8
napari-svg==0.2.1
-e git+https://github.com/AFM-SPM/napari-TopoStats.git@b30d862865188363cfecac500e40e8e192744e7f#egg=napari_TopoStats
nest-asyncio==1.6.0
networkx==3.5
npe2==0.7.9
numba==0.62.1
numcodecs==0.16.3
numpy==2.0.2
numpydoc==1.9.0
numpyencoder==0.3.2
openpyxl==3.1.5
opt_einsum==3.4.0
optree==0.17.0
packaging @ file:///C:/miniconda3/conda-bld/packaging_1761049096285/work
pandas==2.3.3
parso==0.8.5
partd==1.4.2
pillow==11.1.0
Pint==0.25
platformdirs==4.5.0
pluggy==1.6.0
pooch==1.8.2
prompt_toolkit==3.0.52
propcache==0.4.1
protobuf==6.33.0
psutil==5.9.8
psygnal==0.15.0
pure_eval==0.2.3
pyconify==0.2.1
pydantic==2.12.3
pydantic-compat==0.1.2
pydantic_core==2.41.4
Pygments @ file:///C:/b/abs_e4bg5vh5j_/croot/pygments_1744667628203/work
PyOpenGL==3.1.10
pyparsing==3.2.5
pyproject-api==1.10.0
pyproject_hooks==1.2.0
PyQt5==5.15.11
PyQt5-Qt5==5.15.2
PyQt5_sip==12.17.1
pyspm==0.6.3
pytest @ file:///C:/miniconda3/conda-bld/pytest_1758125281680/work
python-dateutil==2.9.0.post0
pytz==2025.2
pywin32==311
PyYAML==6.0.3
pyzmq==27.1.0
qtconsole==5.7.0
QtPy==2.4.3
referencing==0.37.0
requests==2.32.5
rich==14.2.0
roman-numerals-py==3.1.0
rpds-py==0.28.0
ruamel.yaml==0.18.16
ruamel.yaml.clib==0.2.14
schema==0.7.8
scikit-image==0.25.2
scikit-learn==1.7.2
scipy==1.16.3
seaborn==0.13.2
shellingham==1.5.4
six==1.17.0
skan==0.13.0
snoop==0.6.0
snowballstemmer==3.0.1
Sphinx==8.2.3
sphinxcontrib-applehelp==2.0.0
sphinxcontrib-devhelp==2.0.0
sphinxcontrib-htmlhelp==2.1.0
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==2.0.0
sphinxcontrib-serializinghtml==2.0.0
stack-data==0.6.3
superqt==0.7.6
tensorboard==2.20.0
tensorboard-data-server==0.7.2
tensorflow==2.20.0
termcolor==3.2.0
threadpoolctl==3.6.0
tifffile==2025.10.16
tomli_w==1.2.0
toolz==1.1.0
topoly==1.1.0
topostats==2.3.2rc1
tornado==6.5.2
tox==4.32.0
tqdm==4.67.1
traitlets==5.14.3
triangle==20250106
typer==0.20.0
typing-inspection==0.4.2
typing_extensions==4.15.0
tzdata==2025.2
urllib3==2.5.0
virtualenv==20.35.4
vispy==0.15.2
wcwidth==0.2.14
Werkzeug==3.1.3
win32_setctime==1.2.0
wrapt==2.0.0
yarl==1.22.0
zarr==3.1.3
zipp==3.23.0

### PR DESCRIPTION
**Pull request by ns-rse**
Closes #1246

When instantiating the module we extract the release and commit to report in the logs and YAML file that are reported.

This was done by splitting the string, development versions will have the from `#.#.#dev###+g<hash>` and released
versions do not contain the `dev###+g<hash>` component to extract.

We now switch to using the [packaging.version](https://packaging.pypa.io/en/stable/version.html) module to check if we
this is a `dev` (`#.#.#dev###+g<hash>`) or `prerelease` (`#.#.#rc#`) version and set the `TOPOSTATS_BASE_VERSION` and
`TOPOSTATS_COMMIT` conditional on that.

### CODE REVIEW

File: topostats/processing.py
Comment (tobyallwood) [inline topostats/processing.py]: This all makes sense to me, assuming it's fine to still have the `TopoStats Commit:` line but leave it empty rather than removing the line from outputs then?

**Comment by ahobbs7** [comment]
Seems to be working now for me, thanks Neil!

**Comment by ns-rse** [comment]
Thanks for checking @ahobbs7 although I suspect you've tested against the `main` branch which is not where the problem occurs, its with released versions on PyPI which don't carry a `commit`-hash in the version so we'll only know once `v2.3.2rc2` has been released and can be tested.

I'll be jumping through the release hoop shortly.
